templates/aws-stack.yml

```diff
@@ -810,7 +821,7 @@ Resources:
               SpotOptions:
                 MaxPrice: !Ref SpotPrice
             - !Ref "AWS::NoValue"
-          ImageId: !If [ HasImageId, !Ref ImageId, !If [ HasImageIdParameter, !GetAtt ImageIdParameterStack.Outputs.ImageId, !FindInMap [ AWSRegion2AMI, !Ref 'AWS::Region', !Ref InstanceOperatingSystem ] ] ]
+          ImageId: !If [ HasImageId, !Ref ImageId, !If [ HasImageIdParameter, !GetAtt ImageIdParameterStack.Outputs.ImageId, !If [ UseWindowsAgents, !FindInMap [ AWSRegion2AMI, !Ref 'AWS::Region', 'windows'], !If [ UsingArmInstances, !FindInMap [ AWSRegion2AMI, !Ref 'AWS::Region', 'linuxarm'], !FindInMap [ AWSRegion2AMI, !Ref 'AWS::Region', 'linux'] ] ] ] ]
```

Comment:
:vomiting_face:

Reply:
maybe at this point we should multi-line this so it's easier to reason about?

Reply:
Yer, good call. I'll make that change